src/formatDate.js

```diff
@@ -9,6 +9,81 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  let newDate = '';
+  let fromCharacter = 0;
+  let month;
+  let day;
+  let year;
+  let yearFormat;
+
+  fromFormat.forEach((element) => {
+    const toCharacter = fromCharacter + element.length;
+    const separator = fromFormat[fromFormat.length - 1];
+    const currentCharacter = fromCharacter + element.length + separator.length;
+
+    if (element === 'MM') {
+      month = date.slice(fromCharacter, toCharacter);
+      fromCharacter = currentCharacter;
+    }
+
+    if (element === 'DD') {
+      day = date.slice(fromCharacter, toCharacter);
+      fromCharacter = currentCharacter;
+    }
+
+    if (element === 'YYYY') {
+      year = date.slice(fromCharacter, toCharacter);
+      fromCharacter = currentCharacter;
+      yearFormat = element;
+    }
+
+    if (element === 'YY') {
+      year = date.slice(fromCharacter, toCharacter);
+      fromCharacter = currentCharacter;
+      yearFormat = element;
+    }
+  });
+
+  toFormat.forEach((element, index) => {
+    if (element === 'DD') {
+      newDate += day;
+    }
+
+    if (element === 'MM') {
+      newDate += month;
+    }
+
+    if (element === 'YY') {
+      if (yearFormat === 'YY') {
+        newDate += year;
+      } else if (yearFormat === 'YYYY') {
+        newDate += year.slice(-2);
+      }
+    }
+
+    if (element === 'YYYY') {
+      if (yearFormat === 'YYYY') {
+        newDate += year;
+      } else if (yearFormat === 'YY') {
+        if (year < '30') {
+          newDate += `${20}${year}`;
+        } else {
+          newDate += `${19}${year}`;
+        }
+      }
+    }
+
+    if (
```

Comment:
The condition to append the separator is not robust. It assumes there will always be a next element in the `toFormat` array, which might not be the case for the last element. Additionally, it does not account for the possibility of the last element in the `toFormat` array being a separator. Refactor this logic to correctly handle the addition of separators between date components.